### ISSUE TITLE
[java] TooManyMethods triggers for JUnit tests

### ISSUE THREAD

**Issue by mcandre** [pmd/pmd#1611]
**Affects PMD Version:** 

3.9+

**Rule:**

TooManyMethods

**Description:**

This rule can trigger when a JUnit test is thorough, featuring multiple tests. As a workaround, I am disabling this rule in my rulesets.xml:

```xml
<rule ref="category/java/design.xml">
    <exclude name="TooManyMethods" />
</rule>
```

**Running PMD through:** *[CLI | Ant | Maven | Gradle | Designer | Other]*

Gradle

**Comment by Fernal73** [comment]
```
//ClassOrInterfaceDeclaration/ClassOrInterfaceBody
     [
      count(./ClassOrInterfaceBodyDeclaration/MethodDeclaration[
         not (
                starts-with(@Name,'get')
                or
                starts-with(@Name,'set')
                or
                starts-with(@Name,'is')
            )
      ]) > $maxmethods
   ]
```

The rule currently excludes getters, setters and is methods. Methods such as 'are', 'test' and having annotations like @Test signifying a unit test could be added so that test classes don't trigger this rule.

**Comment by Fernal73** [comment]
> ```
> //ClassOrInterfaceDeclaration/ClassOrInterfaceBody
>      [
>       count(./ClassOrInterfaceBodyDeclaration/MethodDeclaration[
>          not (
>                 starts-with(@Name,'get')
>                 or
>                 starts-with(@Name,'set')
>                 or
>                 starts-with(@Name,'is')
>             )
>       ]) > $maxmethods
>    ]
> ```
> 
> The [rule](https://pmd.github.io/latest/pmd_rules_java_design.html#toomanymethods) currently excludes getters, setters and is methods.
> Methods such as 'are', 'test' and having annotations like @test signifying a unit test could be added so that test classes don't trigger this rule.

Another option is to use Nested classes like JUnit5 provides.
